src/code/z_room.c

```diff
@@ -119,44 +119,44 @@ void func_80095D04(GlobalContext* globalCtx, Room* room, u32 flags) {
     spA4 = spB8;
 
     ASSERT(polygon2->num <= SHAPE_SORT_MAX, "polygon2->num <= SHAPE_SORT_MAX", "../z_room.c", 317);
-    sp78 = polygonDlist;
-
-    for (sp9C = 0; sp9C < polygon2->num; sp9C++, polygonDlist++) {
-        sp90.x = polygonDlist->pos.x;
-        sp90.y = polygonDlist->pos.y;
-        sp90.z = polygonDlist->pos.z;
-        SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->viewProjectionMtxF, &sp90, &sp84, &sp80);
-        if (-(f32)polygonDlist->unk_06 < sp84.z) {
-            temp_f2 = sp84.z - polygonDlist->unk_06;
+    polygonDlistFirst = polygonDlist;
+
+    for (i = 0; i < polygon2->num; i++, polygonDlist++) {
+        pos.x = polygonDlist->pos.x;
+        pos.y = polygonDlist->pos.y;
+        pos.z = polygonDlist->pos.z;
+        SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->viewProjectionMtxF, &pos, &projectedPos, &projectedW);
+        if (-(f32)polygonDlist->unk_06 < projectedPos.z) {
+            temp_f2 = projectedPos.z - polygonDlist->unk_06;
             if (temp_f2 < globalCtx->lightCtx.fogFar) {
-                phi_v0 = spB4;
                 spA4->unk_00 = polygonDlist;
                 spA4->unk_04 = temp_f2;
-                if (phi_v0 == NULL) {
+                iter = spB4;
+                if (spB4 == NULL) {
```

Comment:
Pretty minor but I feel like this check is intended to be about `iter` since it's followed by a do while loop on iter != NULL. Is there a particular reason to change it?

Reply:
That's my bad, I didn't realize the do while did expect `iter` initialized. I thought the assignment was a fake match basically. I'll revert the conditional